fuel/src/main/kotlin/com/github/kittinunf/fuel/core/interceptors/RedirectionInterceptor.kt

```diff
@@ -1,15 +1,13 @@
 package com.github.kittinunf.fuel.core.interceptors
 
-import com.github.kittinunf.fuel.Fuel
-import com.github.kittinunf.fuel.core.Encoding
-import com.github.kittinunf.fuel.core.FuelError
-import com.github.kittinunf.fuel.core.Request
-import com.github.kittinunf.fuel.core.Response
+import com.github.kittinunf.fuel.core.*
+import java.net.MalformedURLException
+import java.net.URL
 import javax.net.ssl.HttpsURLConnection
 
 class RedirectException() : Exception("Redirection fail, not found URL to redirect to")
 
-fun redirectResponseInterceptor() =
+fun redirectResponseInterceptor(manager: FuelManager) =
```

Comment:
It looks fine to me. 
- If the `interceptor` is removed from the `manager` then the `interceptor` will become garbage even if it holds a reference to the `manager`
- If the `manager` is no longer referenced both will become garbage even if there is a reference cycle.

The java garbage collector has no problem with cycles, unlike what I believe is the case with iOS or Python. If this is your background maybe this stackoverflow question will make it more clear: http://stackoverflow.com/questions/1910194/how-does-java-garbage-collection-work-with-circular-references